docs/grammar/SolidityParser.g4

```diff
@@ -409,10 +411,10 @@ inlineArrayExpression: LBrack (expression ( Comma expression)* ) RBrack;
 /**
  * Besides regular non-keyword Identifiers, some keywords like 'from' and 'error' can also be used as identifiers.
  */
-identifier: Identifier | From | Error | Revert | Global;
+identifier: Identifier | From | Error | Revert | Global | Suffix;
 
-literal: stringLiteral | numberLiteral | booleanLiteral | hexStringLiteral | unicodeStringLiteral;
-booleanLiteral: True | False;
+literal: (stringLiteral | numberLiteral | booleanLiteral | hexStringLiteral | unicodeStringLiteral) identifier?;
```

Comment:
While at it, we should also rename `NumberUnit` to `Subdenomination`. We use "subdenomination" in the code and the grammar seems to be the only place where a "number unit" is used. In the docs we don't have a common name and refer to some of them as [ether units](https://docs.soliditylang.org/en/latest/units-and-global-variables.html#ether-units) and others as [time units](https://docs.soliditylang.org/en/latest/units-and-global-variables.html#time-units).

Reply:
sounds good